### DISCUSSION TITLE
Hotkeys enabled, but all hotkey textbox are grey with message "You can only edit hotkeys while they are disabled."

### DISCUSSION THREAD

**Discussion by oeloo** [AutoDarkMode/Windows-Auto-Night-Mode#697] [General]
Hi, I cannot have a 'Toggle theme hotkey' because the textbox is gray:
![image](https://user-images.githubusercontent.com/98819304/227219261-19395fe1-80b0-418d-a28d-82abee3f5539.png)

Is it saying "You can only edit hotkeys while they are disabled.", but what does mean ? but nothing is disabled 🤔

**Comment by Armin2208** [comment]
it means you need to disable hot keys to enable the boxes. Only if the toggle is disabled you can set your own hot keys.

**Comment by oeloo** [comment]
When 'Enable system-wide hotkeys' is Off, when I add a shortcut for 'Toggle theme hotkey', this shortcut does not work at all.
I have tried multiple times, whatever the shortcut I enter, this shortcut does not toggle anything.

When 'Enable system-wide hotkeys' is On, what is the system-wide hotkey for 'Toggle theme hotkey' ?

**Comment by Armin2208** [comment]
You turn the Enable system-wide hotkeys toggle off, so you set your keyboard shortcuts. After you set them into the boxes, you activate the toggle Enable system-wide hotkeys again. Your hotkeys that you put into the boxes now work.

**Comment by oeloo** [comment]
None of AutoDark shortcut works on our PC :
![image](https://user-images.githubusercontent.com/98819304/228305154-44a29f78-d68b-40c5-8f05-251d62a5dbc0.png)
whatever the shortcut we enter it does not work. Furthermore I can guarantee that none of 6 shortcuts work.
I guess it is only working on English keyboard. I have a French keyboard (Azerty), and use Windows 11 2H22.

**Comment by Armin2208** [comment]
Look you set Toggle theme hotkey to "Ctrl + Shift + T". Now you need to enable the hotkey toggle to gray out all boxes. After you enabled it, it should work. Or?

**Reply by oeloo** [reply]
> Look you set Toggle theme hotkey to "Ctrl + Shift + T". Now you need to enable the hotkey toggle to gray out all boxes. After you enabled it, it should work. Or?

Look at the screenshot I had sent it my post, I had already done that.
But shortcut does not work !

**Reply by Spiritreader** [reply]
>When 'Enable system-wide hotkeys' is Off, when I add a shortcut for 'Toggle theme hotkey', this shortcut does not work at all.

Yes, that is exactly how this is supposed to work. If you turn it off, it is off. If you turn it on, it is on. But you still need to configure hotkeys as you have done correctly in the latest screenshot.

It is only possible to **edit** hotkeys when they are disabled / turned off / the toggle "Enable system-wide hotkeys" is set to "Off".
At that point the boxes will no longer be grayed out because you are now allowed to edit them.

It is not possible to **edit** hotkeys when the "Enable system-wide hotkeys" toggle is "On", because when an application listens for hotkeys, and you modify them *while* it is "On", collisions appear.

If you want hotkeys to be recognized / working, you need to put the "Enable system-wide hotkeys" toggle to "On" **after** you have configured them when they were "Off".

It is only possible to **use** hotkeys when the "Enable system-wide hotkeys" setting is "On".

> I guess it is only working on English keyboard. 

No.

After trying this, if your hotkeys are still not working after following the exact steps in the video, something is blocking hotkeys on your system, or there is an issue with registering them, in which case we would need log files.

https://user-images.githubusercontent.com/18438899/229124724-1d8dca73-0b46-4ca2-a627-10367e6940aa.mp4

**Reply by oeloo** [reply]
Thank you for the explanation it is now working.

But I have to say: what a complicate and unusual UI for defining a keyboard short.
I had never seen this.
You can have a look at how application like Ditto, Flow Launcher, Fluent Search , manage the same needs: creating shorcuts in Windows. You will see it is much more user friendly. They are also opensource, so you can have a look how it is done.

**Reply by Spiritreader** [reply]
> Thank you for the explanation it is now working.
> 
> But I have to say: what a complicate and unusual UI for defining a keyboard short. I had never seen this. You can have a look at how application like Ditto, Flow Launcher, Fluent Search , manage the same needs: creating shorcuts in Windows. You will see it is much more user friendly. They are also opensource, so you can have a look how it is done.

It's... not exactly rocket science to figure this out. There's literally only two options, on or off.
If somebody wants to make it prettier, sure, but I don't see any reason to.

**Reply by oeloo** [reply]
> It's... not exactly rocket science to figure this out. There's literally only two options, on or off. If somebody wants to make it prettier, sure, but I don't see any reason to.

Don't hear your user. Do not look at what other dev/UX have done. This is how to end up with such bad UI design.

**Reply by Spiritreader** [reply]
> > It's... not exactly rocket science to figure this out. There's literally only two options, on or off. If somebody wants to make it prettier, sure, but I don't see any reason to.
> 
> Don't hear your user. Do not look at what other dev/UX have done. This is how to end up with such bad UI design.

Why should I bother implementing something for someone for free when I'm happy with the current state.
ADM is not a paid or subscription based product, it's FOSS.
And in the spirit of that we welcome contributors that want to improve upon this design, but it's not a priority because it would require time and resources that simply aren't there for a hobby project.

**Comment by oeloo** [comment]
I fully recognize the effort you do in contributing to Auto Dark Mode. Thank you. It does not give the power to overlook the user though.
Users and concrete feedbacks should still be taken into account at the level the community can do of course. Maybe by adding it in the tool feature wishlist, so that future potential dev could know about this not so user-friendly shotcut entry and modify it.

**Reply by Spiritreader** [reply]
> I fully recognize the effort you do in contributing to Auto Dark Mode. Thank you. It does not give the power to overlook the user though. Users and concrete feedbacks should still be taken into account at the level the community can do of course. Maybe by adding it in the tool feature wishlist, so that future potential dev could know about this not so user-friendly shotcut entry and modify it.

The user is not overlooked in this scenario. You're free to submit a feature suggestion via github issue and outline how you would like hotkey management changed.
It's just that neither maintainers nor contributors have any obligation to follow up on it.
I'm personally not interested in it, but maybe someone else is.

In theory we do have the power to overlook non-paid suggestions and bug reports by making support / issue reports paid only, which has already happened with some open source software projects. However we currently do not want that.

**Reply by YKPDang** [reply]
I think it absolutely gives them the power to "overlook" the users, it gives them the power to do whatever they want to do. No one is paying them to do this and they do this in their free time. If you are unhappy with how something works, instead of sounding like an entitled _paying customer_, you could ask them nicely if they'd consider changing something to make it (in your eyes) more user friendly.

They are maintaining this for free, in their own free time. Sure, they have now a big community and a large user base, but that still doesn't mean anything. If you want change, either do it yourself, ask _nicely_ through the appropriate channels, or keep it to yourself. You're essentially saying "You, being maintainer of ADM, have no power to decide which features should be developed for ADM"

**Reply by oeloo** [reply]
It is not because you can, that you should do. It is too different things.

I would have appreciated instead a suggestion to create an issue about this. I have understood, that you will give no interest to it.  And to be honest, I don't care ever, because now that I know how to manage this weird UI design, I do not need it either, so it is not for me, but for future AutoDarkMode users.
But I am sure of one thing: it is such an unusual UI design, that most users will not use this hotkey system because of the way the UI is designed. Opensource project do not belong to the current maintainer, they belong to the society. Offering easy possibility to do a hotkey switch for the theme is serving this purpose.

**Reply by YKPDang** [reply]
I disagree that it is an unusual UI design. I have encountered this kind of hotkey setting many times, although admittedly, off the top of my head I can't name a program.

While I agree that an OSS project with an active user base has a certain _societal_ "obligation" to listen to feedback, society absolutely does not own the project. 

I don't think there are many people struggling with the hotkey functionality - you are the first one to bring it up and then also expect the maintainers to change it according to your wishes. That is not how any OSS works, unless you pay them. The maintainers always decide what they want to spend their time working on and if there is a UX/UI issue that is reportedly experienced by just one person, is it really worth it for them to change the logic of that feature? It might be worth it if they agree with you that the current way is bad and could affect more users.

In my opinion, the message "you can only edit hotkeys while hotkeys are disabled" is displayed right beneath the hotkey toggle, so to me it's clear that I have to switch that toggle to disable hotkeys in order for me to edit hotkeys.

I just saw that you opened an issue to call for a change in this functionality, and that's good! I'm sorry for piling onto you here, but I've just seen one too many FOSS users who themselves contributed nothing but came across as very entitled.

**Reply by oeloo** [reply]
I hope that this tiny contribution, giving feedback experience from a none Auto Dark Mode expert, helps though I understand and respect it is not your priority.
As I have already set my hotkey and it only has to be set once, I will certainly never see this setting UI again. So I have not written neither my post nor the issue for me. I have done it because my experience tells me that most not expert Auto Dark Mode users will not understand the hotkey process.

**Comment by Jay-o-Way** [comment]
Hey, just wanted to say that there's a new beta version released, with a completely new UX. This issue is resolved in the new version. 🙂
